service/src/services/domain/builders/community-external-invitation-created/community.external.invitation.created.notification.builder.ts

```diff
@@ -0,0 +1,118 @@
+import { Injectable, Inject } from '@nestjs/common';
+import { NotificationEventType } from '@alkemio/notifications-lib';
+import { INotificationBuilder } from '@core/contracts';
+import { ExternalUser, User } from '@core/models';
+import { CommunityExternalInvitationCreatedEventPayload } from '@alkemio/notifications-lib';
+import {
+  AlkemioUrlGenerator,
+  NotificationBuilder,
+  RoleConfig,
+} from '../../../application';
+import { NotificationTemplateType } from '@src/types';
+import { EmailTemplate } from '@common/enums/email.template';
+import { CommunityExternalInvitationCreatedEmailPayload } from '@common/email-template-payload';
+import { ALKEMIO_URL_GENERATOR } from '@src/common/enums/providers';
+
+@Injectable()
+export class CommunityExternalInvitationCreatedNotificationBuilder
+  implements INotificationBuilder
+{
+  constructor(
+    @Inject(ALKEMIO_URL_GENERATOR)
+    private readonly alkemioUrlGenerator: AlkemioUrlGenerator,
+    private readonly notificationBuilder: NotificationBuilder<
+      CommunityExternalInvitationCreatedEventPayload,
+      CommunityExternalInvitationCreatedEmailPayload
+    >
+  ) {}
+
+  build(
+    payload: CommunityExternalInvitationCreatedEventPayload
+  ): Promise<NotificationTemplateType[]> {
+    const roleConfig: RoleConfig[] = [
+      {
+        role: 'inviter',
+        emailTemplate: EmailTemplate.COMMUNITY_EXTERNAL_INVITATION_INVITER,
+      },
+      {
+        role: 'invitee',
+        emailTemplate: EmailTemplate.COMMUNITY_EXTERNAL_INVITATION_INVITEE,
+      },
+    ];
+
+    const templateVariables = {
+      inviterID: payload.triggeredBy,
+      spaceID: payload.journey.spaceID,
+      challengeID: payload.journey.challenge?.id ?? '',
+      opportunityID: payload.journey.challenge?.opportunity?.id ?? '',
+    };
+
+    // const externalUsers = payload.invitees.map();
+
+    return this.notificationBuilder.build({
+      payload,
+      eventUserId: payload.triggeredBy,
+      roleConfig,
+      templateType: 'community_external_invitation_created',
+      templateVariables,
+      templatePayloadBuilderFn: this.createTemplatePayload.bind(this),
+      externalUsers: [
+        { email: 'john.wick@mail.io', firstName: '', lastName: '' },
```

Comment:
I am quite confident you shouldn't be sending emails to John Wick. Can you please fix the payload generation?

Reply:
Yes, 100%.